Videomaker: Fix category trailing slash

mikachan opened the pull request:
Co-authored-by: @danieldudzic

<!-- Thanks for contributing to our free themes! Please provide as much information as possible with your Pull Request by filling out the following - this helps make reviewing much quicker! -->

#### Changes proposed in this Pull Request:
This PR fixes the trailing slash if a post does not have categories.

![image](https://user-images.githubusercontent.com/1645628/138257692-b7ce8567-188b-47ef-8b5b-a2abc15040bc.png)

This also removes the tag styles if tags are added to the `post-meta` group:

![image](https://user-images.githubusercontent.com/1645628/138257854-32e6fabe-cae2-457a-81fb-e8665af36c2a.png)

#### Related issue(s):
Closes #4882

mikachan committed:
Videomaker: fix category trailing slash

Co-authored-by: @danieldudzic

MaggieCabrera commented on videomaker/sass/_meta.scss:
```diff
@@ -4,12 +4,29 @@ div.post-meta {
+			-webkit-mask-image: none;
+			mask-image: none;
+			width: auto;
+			height: auto;
```
Do we need this because you are using before for the slash? or do we need this anyway? Beacause I'm thinking, a different way of doing this would be something like: `.post-meta p:last-child { display: none; }` it will only happen when the post-terms blocks is not present because afaik when it's not there there is no empty div.

mikachan replied:
Oh yeah that sounds good! All this was because I was using the before for the slash. Let me try this out now.

mikachan replied:
Ah, it was also added to handle multiple post meta blocks being added by the user (or the user just moving these blocks around in general). At the moment, I think users would need to manually add a slash themselves between each item.

MaggieCabrera replied:
I'm going to start a PR to move the icons outside of the general post meta styles

MaggieCabrera replied:
[This](https://github.com/Automattic/themes/pull/4891) removes all the extra CSS from the icons if the class is not applied, should reduce the needed CSS here

MaggieCabrera replied:
Also, Quadrat does the same thing with a dot as a separator, so maybe you want to grab inspiration from that.

mikachan replied:
I've updated this so Videomaker is tweaking the styles in the same way as Quadrat. I think it looks good, the styles definitely look better!

This means that this change needs [your PR](https://github.com/Automattic/themes/pull/4891) before it's merged.

MaggieCabrera replied:
Just did! now I think this needs a rebase

mikachan replied:
Woop, this is now up to date with trunk and still working 😁

mikachan committed:
Videomaker: simplify post meta styles

mikachan committed:
Merge branch 'trunk' into fix/4882-videomaker-trailing-slash

scruffian commented on videomaker/sass/_meta.scss:
```diff
@@ -4,4 +4,12 @@ div.post-meta {
+	> * {
+		&:not(:first-child):before {
+			color: var(--wp--custom--color--foreground);
+			content: "/";
```
what happens with i18n and RTL if we take this approach? I guess we had that problem before :)

scruffian commented on videomaker/block-template-parts/post-meta.html:
```diff
@@ -1,10 +1,12 @@
-<!-- wp:group {"className":"post-meta","layout":{"type":"flex"}} -->
-<div class="wp-block-group post-meta">
+<!-- wp:group {"className":"post-meta","layout":{"type":"flex"},"style":{"spacing":{"padding":{"top":"1em","bottom":"1em"}}}} -->
```
Should we use the block-gap variable here?

mikachan replied:
Yeah good idea! I think variables usually make more sense than fixed values.

scruffian committed:
Fix RTL styles

scruffian commented:
In RTL this did look like this
<img width="389" alt="Screenshot 2021-10-22 at 09 55 41" src="https://user-images.githubusercontent.com/275961/138425687-d9b3a2a5-1916-48e6-83a0-b0414fa1f637.png">

With the changes I just made it now looks like this:
<img width="506" alt="Screenshot 2021-10-22 at 09 55 52" src="https://user-images.githubusercontent.com/275961/138425731-4c2c64e4-2f2f-4116-bc89-78fa6cb9339d.png">

mikachan committed:
Use gap variable for padding values

scruffian committed:
Use the block-gap variable and add a fallback

scruffian commented:
@mikachan I updated this to use the --wp--style--block-gap variable as that's what we use for spaces between blocks. I also added 1em as a fallback.

scruffian reviewed (approved):
LGTM if you're happy with my changes

mikachan commented:
> @mikachan I updated this to use the --wp--style--block-gap variable as that's what we use for spaces between blocks. I also added 1em as a fallback.

Ah thanks, I didn't realise. These changes look good to me, thanks for your help!
